main.c

```diff
@@ -54,14 +54,26 @@ usage(void)
 {
 
 	fprintf(stderr,
-	    "usage: scrypt {enc | dec | info} [-f] [-M maxmem]"
-	    " [-m maxmemfrac] [-P]\n"
-	    "              [-t maxtime] [-v] [--passphrase method:arg]"
-	    " infile [outfile]\n"
+	    "usage: scrypt {enc | dec | info} [-f] [--logN value] [-M maxmem]\n"
+	    "              [-m maxmemfrac] [-P] [-p value] [-r value]"
+	    " [-t maxtime] [-v]\n"
+	    "              [--passphrase method:arg] infile [outfile]\n"
 	    "       scrypt --version\n");
 	exit(1);
 }
 
+/* Parse a numeric optarg within a GETOPT context.  (Requires ch and optarg.) */
+#define GETOPT_PARSENUM_WITHIN_UNSIGNED(var, min, max) do {		\
+	if (PARSENUM((var), optarg, (min), (max))) {			\
+		if (errno == ERANGE) {					\
+			warn0("%s must be between %zu and %zu"		\
```

Comment:
You're printing these as `size_t` variables, but you're not casting them to that type.

Reply:
Argh.

I'm tempted to do a whole parsenum-style "print as double, intmax_t, or uintmax_t" macro, but I can't find enough instances of "parse a number within this range" in our code to make it IMO worth it.

Since unattached numbers are intepreted as `int`, I renamed this to `GETOPT_PARSENUM_WITHIN_INT`.

Reply:
Well, I guess you could do that.  Although I would still prefer to see explicit casts inside the printf.

But I was more thinking of casting to `uintmax_t` and using `%ju`...

Reply:
That was my first thought, but then I figured that you'd say it was over-engineered.

... ok, updated.